ql/src/test/queries/clientpositive/udf_date_format.q

```diff
@@ -78,3 +78,16 @@ select date_format("2015-04-08 10:30:45","yyyy-MM-dd HH:mm:ss.SSS z");
 --julian date
 set hive.local.time.zone=UTC;
 select date_format("1001-01-05","dd---MM--yyyy");
+
+--dates prior to 1900
+set hive.local.time.zone=Asia/Bangkok;
+select date_format('1400-01-14 01:01:10.123', 'yyyy-MM-dd HH:mm:ss.SSS z');
+select date_format('1800-01-14 01:01:10.123', 'yyyy-MM-dd HH:mm:ss.SSS z');
+
+set hive.local.time.zone=Europe/Berlin;
+select date_format('1400-01-14 01:01:10.123', 'yyyy-MM-dd HH:mm:ss.SSS z');
+select date_format('1800-01-14 01:01:10.123', 'yyyy-MM-dd HH:mm:ss.SSS z');
+
+set hive.local.time.zone=Africa/Johannesburg;
+select date_format('1400-01-14 01:01:10.123', 'yyyy-MM-dd HH:mm:ss.SSS z');
```

Comment:
Do we have tests for other formats (to ensure DateTimeFormat doesn't break anything)? Also need update of wiki doc.

Reply:
All the existing tests with SimpleDateFormat Formatter is passing except the milliseconds change which I have mentioned in my comment.

Reply:
Once this patch is merged I will update the Hive wiki as well

Reply:
+1 to @sankarh comments. Existing test cases are fine. Could please add some more time zone based test cases.

Reply:
done